src/components/Popover/src/Popover.vue

```diff
@@ -0,0 +1,278 @@
+<template>
+	<div style="display: contents;">
+		<slot
+			name="tether"
+			v-bind="actionAPI"
+		/>
+
+		<portal
+			v-if="actionAPI.isOpen"
+			:selector="popoverApi.targetSelector"
+		>
+			<slot name="content" />
+		</portal>
+	</div>
+</template>
+
+<script>
+import { throwError } from '@square/maker/utils/debug';
+import assert from '@square/maker/utils/assert';
+import { Portal } from '@linusborg/vue-simple-portal';
+import { v4 as uuid } from 'uuid';
+import { PopoverConfigKey, PopoverAPIKey } from './keys';
+
+const MAX_TETHER_VNODE = 1;
+
+const getMinWidth = (minWidth, tetherMinWidth, reference) => {
+	if (!tetherMinWidth) {
+		return minWidth;
+	}
+
+	return minWidth || reference.offsetWidth;
+};
+
+const createPopperConfig = ({
+	placement,
+	offset,
+	tetherMinWidth,
+	minWidth,
+}) => ({
+	placement,
+	modifiers: [
+		{
+			name: 'offset',
+			options: { offset },
+		},
+		{
+			name: 'preventOverflow',
+			options: {
+				padding: 24,
+				altBoundary: true,
+			},
+		},
+		{
+			name: 'minWidth',
+			enabled: tetherMinWidth || !!minWidth,
+			phase: 'beforeWrite',
+			requires: ['computeStyles'],
+			fn({ state: { styles, rects } }) {
+				const popoverMinWidth = getMinWidth(minWidth, tetherMinWidth, rects.reference);
+				styles.popper.minWidth = `${popoverMinWidth}px`;
+			},
+			effect({ state: { elements } }) {
+				const popoverMinWidth = getMinWidth(minWidth, tetherMinWidth, elements.reference);
+				elements.popper.style.minWidth = `${popoverMinWidth}px`;
+			},
+		},
+	],
+});
+
+export default {
+	name: 'Popover',
+
+	components: {
+		Portal,
+	},
+
+	inject: {
+		popoverApi: {
+			from: PopoverAPIKey,
+			default: () => throwError('No popover layer detected', 'Popover'),
+		},
+
+		popoverConfig: {
+			from: PopoverConfigKey,
+			default: {
+				config: undefined,
+			},
+		},
+	},
+
+	props: {
+		/**
+		 * Starting position of the popover. Not guaranteed if too close to overflow.
+		 */
+		placement: {
+			type: String,
+			default: 'bottom-start',
+			validator: (placement) => [
+				'auto', 'auto-start', 'auto-end',
+				'top', 'top-start', 'top-end',
+				'right', 'right-start', 'right-end',
+				'bottom', 'bottom-start', 'bottom-end',
+				'left', 'left-start', 'left-end',
+			].includes(placement),
+		},
+
+		/**
+		 * Distance from tether element
+		 */
+		distanceOffset: {
+			type: Number,
+			default: 8,
+		},
+
+		/**
+		 * Offset from base position (Y for left/right placement, X for top/bottom placement)
+		 */
+		skiddingOffset: {
+			type: Number,
+			default: 0,
+		},
+
+		/**
+		 * Absolute min width of popover, overrides tetherMinWidth
+		 */
+		minWidth: {
```

Comment:
I think this could also be removed. Don't we want to calculate the width based on the element we're wrapping? There'd be a static min-width if the element is under a certain threshold. Otherwise the popover container would be the same width as the element we're wrapping.